src/lib/decorators/types.js

```diff
@@ -1,10 +1,17 @@
 // @flow
 
-export type Decorator = (
-  target: Object,
+export type Decorator<TType> = (
+  target: TType,
   name: string,
-  descriptor: Object,
-) => Object;
+  descriptor: Descriptor,
+) => Descriptor;
+
+export type Descriptor = {
+    configurable: boolean,
+    enumerable: boolean,
+    value: Function,
```

Comment:
descriptor value, in general, could be not only a function, right? but we will use decorators only for functions?